### DISCUSSION TITLE
Trivy k8s compliance scan error

### DISCUSSION THREAD

**Discussion by Susantade** [aquasecurity/trivy#4607] [Q&A] [answered]
### Question

Hi all,
I am a new trivy user. I am scanning my EKS cluster by trivy k8s compliance. When I scan my whole EKS cluster using trivy, most of compliances results are null. I am unable to understand what this null actually means. Does it mean compliance pass or fail? Can you help me out? I am using trivy version 0.40.0.
My command is:  trivy k8s cluster --compliance=k8s-nsa -f json -o k8s-nsa.json --report all

### Target

Kubernetes

### Scanner

Misconfiguration

### Output Format

JSON

### Mode

Standalone

**Comment by AnaisUrlichs** [comment]
Heya, thank you for reaching out. Could you please explain what you mean by the following?
> compliances results are null

When I run the scan (using your command) on my local kind cluster, I get the a long `Results` array, the beginning of my file looks like this:
```
❯ cat k8s-nsa.json 
{
  "ID": "k8s-nsa",
  "Title": "National Security Agency - Kubernetes Hardening Guidance v1.0",
  "Description": "National Security Agency - Kubernetes Hardening Guidance",
  "Version": "1.0",
  "RelatedResources": [
    "https://www.nsa.gov/Press-Room/News-Highlights/Article/Article/2716980/nsa-cisa-release-kubernetes-hardening-guidance/"
  ],
  "Results": [
    {
      "ID": "1.0",
      "Name": "Non-root containers",
      "Description": "Check that container is not running as root",
      "Severity": "MEDIUM",
```

**Reply by Susantade** [reply]
{
  "ID": "k8s-nsa",
  "Title": "National Security Agency - Kubernetes Hardening Guidance v1.0",
  "Description": "National Security Agency - Kubernetes Hardening Guidance",
  "Version": "1.0",
  "RelatedResources": [
    "https://www.nsa.gov/Press-Room/News-Highlights/Article/Article/2716980/nsa-cisa-release-kubernetes-hardening-guidance/"
  ],
  "Results": [
    {
      "ID": "1.0",
      "Name": "Non-root containers",
      "Description": "Check that container is not running as root",
      "Severity": "MEDIUM",
      "Results": null

**Reply by AnaisUrlichs** [reply]
Is this just the case for "Non-root containers" or for all other checks as well?

**Reply by Susantade** [reply]
Yeah, for others as well

**Reply by itaysk** [reply]
@chen-keinan ?

**Comment by chen-keinan** [comment, marked as answer]
@Susantade yes you are right , `"Results": null` mean `control-check` has `passed`.
I'll explain how it work behind the scene.
The compliance report is backed up by mis-config `rego engine` meaning that for each `control-check` it execute set rego policies to evaluate the `control-check`.
The result will not be `null` only when rego policy check has failed.

I assume it can be improved to be more informative.

**Reply by itaysk** [reply]
Thanks Chen. To add to that:
In a regular Trivy scan the user experience is about finding security issues, so the results are always going to be "problems", and if there are no problems you will not see any results. With compliance reports the user experience is about coverage of the rules in the compliance framework; even if there are no problems you will always see the same results, to show you how you stand in compliance. (cc @AnaisUrlichs )

**Reply by Susantade** [reply]
Thanks @chen-keinan @itaysk for clearing me.
One more doubt when the **successes** field value not be 0(zero)? While "Privileged Containers" rule passed on any kubernetes resource?
```
{
      "ID": "3",
      "Name": "Privileged Containers",
      "Description": "Privileged Pods disable most security mechanisms and must be disallowed.",
      "Severity": "HIGH",
      "Results": [
        {
          "Target": "DaemonSet/kube-proxy",
          "Class": "config",
          "Type": "kubernetes",
          "MisconfSummary": {
            "Successes": 0,
            "Failures": 1,
            "Exceptions": 0
          },
```

**Reply by chen-keinan** [reply]
@Susantade it works the other way around , success mean no failures, Successes field is always `0` when there are failures, if there are no failures `results = null`  (all resources passed check).

table format is more friendly for user.

**Reply by Susantade** [reply]
Thanks @chen-keinan
Got it. Thanks for helping me.
